realm/realm-library/src/main/java/io/realm/RealmList.java

```diff
@@ -170,14 +174,20 @@ private boolean isAttached() {
      * @throws IndexOutOfBoundsException if {@code location < 0 || location > size()}.
      */
     @Override
-    public void add(int location, E object) {
+    public void add(int location, @Nullable E object) {
```

Comment:
IMO if the primitive list support null values, the `LinkView` should support it as well. Otherwise it feels so inconsistent.